src/funcx_common/messagepack/message_types/result.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+import enum
+import typing as t
+import uuid
+
+import pydantic
+
+from .base import Message, meta
+
+
+class ResultErrorCode(enum.Enum):
+    ContainerError = "ContainerError"
+    ManagerLost = "ManagerLost"
+    EndpointGone = "EndpointGone"
+    ProvisioningError = "ProvisioningError"
+
+
+class ResultErrorDetails(pydantic.BaseModel, use_enum_values=True):
+    # is the error related to FuncX internals or not?
+    is_system_error: bool
+    # an enumerated string for the error code
+    code: ResultErrorCode
+    # the user is always supposed to see user_message somewhere
+    user_message: str
```

Comment:
@kylechard @joshbryan-globus Would be good to have your input on this. The idea here is to indicate to the service when a task failed due to an infrastructure failure. This would allow us to trigger retries.

Reply:
I'm have a lot of questions about the semantics, but I'm not opposed to the idea.  Would we always trigger a retry on a system error? Does this include things like Slurm errors or is this only errors that originate with FuncX components?  Is not finding a container a system error?  Is this flag functionally dependent on the code?  E.g. Can `is_system_error` be inferred from the `code` or are there error codes that might be returned that are `is_system_error = True` sometimes and `is_system_error = False` others?  

If this can be inferred from the code, it might be easier to just have the one field and base any retry logic off of `cod` since some codes may have different retry rules.  Alternatively, if the thinking is that this is used primarily to drive retry logic, then maybe calling it `should_retry` would be clearer.

Reply:
Given that a few of us are uncertain about this and at least one of us is strongly against it, I'm removing this.
It can always be added later if it's appropriate. I'll make the change, update tests, and if CI passes that will resolve the last outstanding issue on this PR -- so I will merge.